Strategic/Queen Command.cpp

```diff
@@ -2800,32 +2795,32 @@ void EnemyCapturesPlayerSoldier( SOLDIERTYPE *pSoldier )
 		return;
 	}
 
-	// ATE: Patch fix If in a vehicle, remove from vehicle...
-	TakeSoldierOutOfVehicle( pSoldier );
-
-	HandleMoraleEvent( pSoldier, MORALE_MERC_CAPTURED, pSoldier->sSectorX, pSoldier->sSectorY, pSoldier->bSectorZ );
-
-	// Change to POW....
-	//-add him to a POW assignment/group
-	if( ( pSoldier->bAssignment != ASSIGNMENT_POW )  )
+	if (gStrategicStatus.ubNumCapturedForRescue >= CalculateMaximumPrisonerAmount())
 	{
-		SetTimeOfAssignmentChangeForMerc( pSoldier );
+		SetTimeOfAssignmentChangeForMerc(pSoldier);
+		return;
 	}
 
-	ChangeSoldiersAssignment( pSoldier, ASSIGNMENT_POW );
-	RemoveCharacterFromSquads( pSoldier );
-
-	WORLDITEM			WorldItem;
-	std::vector<WORLDITEM> pWorldItem;
-
-#ifdef JA2UB
-	if (gStrategicStatus.ubNumCapturedForRescue < 3 && (gubQuest[QUEST_HELD_IN_ALMA] == QUESTNOTSTARTED || gubQuest[QUEST_INTERROGATION] == QUESTNOTSTARTED))
-#else
+#ifndef JA2UB

```

Comment:
It'd be great if UB could be gated in this surrendering functionality in a single spot.